lib/mainloop-control.c

```diff
@@ -450,6 +452,63 @@ export_config_graph(ControlConnection *cc, GString *command, gpointer user_data,
   control_connection_send_reply(cc, result);
 }
 
+static void
+control_internal_logs(ControlConnection *cc, GString *command, gpointer user_data, gboolean *cancelled)
+{
+  GString *result = g_string_new("");
+  gchar **arguments = g_strsplit(command->str, " ", 0);
+  AFInterLive status = afinter_get_live_collection_status();
```

Comment:
as I mentioned, in this case, signaling back to the caller a probably bad status would be better implemented in the `afinter_stop_live_collection` and the `afinter_get_size_of_internal_logs` functions

the main reason is that both of them can follow a flow like

- locking properly
- determining a possible invalid status, and 
- unclocking and returning with an invalid status signal, or
- doing their job
- unlocking

this will guarantee that the internal state will not change during the whole process, that is not the case unfortunately using this `afinter_get_live_collection_status()` function

Reply:
@HofiOne : okay, thanks for the insights.